lib/vsc/filesystem/gpfs.py

```diff
@@ -998,3 +999,32 @@ def get_mmhealth_state(self):
         states = res['State']
         comp_entities = ['%s_%s' % ident for ident in zip(states['component'], states['entityname'])]
         return dict(zip(comp_entities, states['status']))
+
+    def xdmod_list_quota(self, devices=None):
+        """ Quota information in xdmod format """
+
+        allquota = self.list_quota(devices)
+        date = datetime.utcnow().isoformat(timespec='seconds')
+        filesys = self.list_filesystems()
+
+        xdmod_list = []
+        for fsname in allquota.keys():
+            mountpoint = filesys[fsname]['defaultMountPoint']
+            for quots in allquota[fsname].values():
+                for qentries in quots.values():
+                    for qinfo in qentries:
+                        entry = {
+                            'resource' : fsname,
+                            'mountpoint' : mountpoint,
+                            'user' : qinfo.name,
+                            'pi' : qinfo.filesetname,
+                            'dt' : date,
+                            'soft_threshold' : qinfo.blockQuota,
+                            'hard_threshold' : qinfo.blockLimit,
+                            'file_count' : qinfo.filesUsage,
+                            'logical_usage' : qinfo.blockUsage,
+                            'physical_usage' : qinfo.blockUsage,
```

Comment:
euhm, we have the mutiplication factors somewhere i think. perhaps use them?

Reply:
we have this in vsc-config / filesystem_info.conf. but this config is nowhere used in this repo and i guess for the better, this is only file system interfacing. what's also missing is eg the mapping of uid to username/fset. for the current scripts, this is done in the parent code path, like vsc-administration. So I think (part of) this code should be moved somewhere else.